### PR TITLE
Replace Flux.Data with MLDatasets

### PR DESCRIPTION
**Pull request by ayush-1506**
WIP: Need to update all examples.

### CODE REVIEW

**Comment by ayush-1506** [comment]
The test is failing because calling `MNIST.traindata()` from MLDatasets for the first time asks the user
```
Do you want to download the dataset from ["https://ossci-datasets.s3.amazonaws.com/mnist/train-images-idx3-ubyte.gz", 
"https://ossci-datasets.s3.amazonaws.com/mnist/train-labels-idx1-ubyte.gz", "https://ossci-datasets.s3.amazonaws.com
/mnist/t10k-images-idx3-ubyte.gz", "https://ossci-datasets.s3.amazonaws.com/mnist/t10k-labels-idx1-ubyte.gz"] to "/home
/runner/.julia/datadeps/MNIST"?
[y/n]
```

Not sure how I can push `y` to input stream here. There also doesn't seem to be a force download option.

File: examples/mnist/mnist.jl [hunk 1]
```diff
@@ -22,8 +22,9 @@ pyplot(size=(600, 300*(sqrt(5)-1)));
 
 # Downloading the MNIST image dataset:
 
-import Flux.Data.MNIST
-images, labels = MNIST.images(), MNIST.labels();
+import MLJDatasets
```
Comment (ablaom) [inline examples/mnist/mnist.jl:25]: I guess this should be `MLDatasets` not `MLJDatasets`

**Comment by ablaom** [comment]
@ayush-1506  Please see https://github.com/JuliaML/MLDatasets.jl/issues/63 for a solution.

**Review by ablaom — COMMENTED** [review]
I expect we need to re-generate mnist.ipynb (by executing mnist.jl). Are you able to do that?

**Comment by ayush-1506** [comment]
@ablaom Yes, updated the notebooks and readme. Would love a review.

**Comment by ablaom** [comment]
@ayush-1506 If I set `execute=true` in the second last line of mnist.jl, to generate a notebook *including cell evaluations*, then I get some issue with the de-serialization.

```julia
[warn | JLSO]: UndefVarError: ##318 not defined
[warn | JLSO]: UndefVarError: ##318 not defined
┌ Error: error when executing notebook based on input file: `~/Dropbox/Julia7/MLJ/MLJFlux/examples/mnist/mnist.jl`
└ @ Literate ~/.julia/packages/Literate/XP0eh/src/Literate.jl:607
ERROR: LoadError: LoadError: MethodError: no method matching Machine(::Vector{UInt8}; cache=true)  
Closest candidates are:
  Machine(!Matched::M, !Matched::AbstractNode...; cache) where M<:MLJModelInterface.Model at /Users/anthony/.julia/packages/MLJBase/gcxd6/src/machines.jl:32
in expression starting at string:1
when executing the following code block

```julia
mach2 = machine(joinpath(DIR, "mnist_machine5.jlso"))
predict_mode(mach2, images[501:503])

Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] execute_block(sb::Module, block::String)
    @ Literate ~/.julia/packages/Literate/XP0eh/src/Literate.jl:720
  [3] execute_notebook(nb::Dict{Any, Any})
    @ Literate ~/.julia/packages/Literate/XP0eh/src/Literate.jl:623
  [4] (::Literate.var"#34#36")()
    @ Literate ~/.julia/packages/Literate/XP0eh/src/Literate.jl:604
  [5] cd(f::Literate.var"#34#36", dir::String)
    @ Base.Filesystem ./file.jl:106
  [6] jupyter_notebook(chunks::Vector{Literate.Chunk}, config::Dict{String, Any})
    @ Literate ~/.julia/packages/Literate/XP0eh/src/Literate.jl:603
  [7] notebook(inputfile::String, outputdir::String; config::Dict{Any, Any}, kwargs::Base.Iterators.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:execute,), Tuple{Bool}}})    
    @ Literate ~/.julia/packages/Literate/XP0eh/src/Literate.jl:540
  [8] top-level scope
    @ ~/Dropbox/Julia7/MLJ/MLJFlux/examples/mnist/mnist.jl:225
  [9] include(fname::String)
    @ Base.MainInclude ./client.jl:444
 [10] top-level scope
    @ REPL[2]:1
 [11] top-level scope
    @ ~/.julia/packages/CUDA/3VnCC/src/initialization.jl:81
in expression starting at /Users/anthony/Dropbox/Julia7/MLJ/MLJFlux/examples/mnist/mnist.jl:225
```

Are you able to reproduce this issue, and if so, investigate? 

 I haven't tried working directly with the notebook and evaluating, just running the Literate script.

**Comment by ablaom** [comment]
Update: This is not an issue if I just run the notebook by hand. I'm going to update the notebook by hand for now, merge this PR and open a separate issue.

**Comment by ablaom** [comment]
@ayush-1506 Thanks!

**Pull request merged**